### ISSUE TITLE
Attestation error: Failed to get the quote

### ISSUE THREAD

**Issue by diegoara96** [canonical/tdx#142]
I have this problem when i try to execute the test_tdx_attest program

```
root@tdx-guest:/usr/share/doc/libtdx-attest-dev/examples# ./test_tdx_attest 
		
		TDX report data

 00000000: 55 37 4a 18 b8 61 7d 1b b9 86 d8 e6 a9 12 16 7c
 00000010: a2 9b 24 2d 87 5e 4a 12 68 9b 1c b8 dd 65 9c 32
 00000020: 9c e6 4b 54 47 c8 70 00 4e 48 e6 f8 5b fd 74 fd
 00000030: 98 99 2a 1f f7 75 32 5f 10 4e 18 ed b4 b4 20 50

Wrote TD Report to report.dat
Failed to get the quote
```

Output of journalctl -u qgsd

```
tee-fhaas@tee-fhaas:~$ journalctl -u qgsd
jun 12 07:36:40 tee-fhaas systemd[1]: Starting qgsd.service - Intel(R) TD Quoting Generation Servic>
jun 12 07:36:40 tee-fhaas usermod[16557]: add 'qgsd' to group 'sgx_prv'
jun 12 07:36:40 tee-fhaas usermod[16557]: add 'qgsd' to shadow group 'sgx_prv'
jun 12 07:36:40 tee-fhaas usermod[16566]: add 'qgsd' to group 'sgx'
jun 12 07:36:40 tee-fhaas usermod[16566]: add 'qgsd' to shadow group 'sgx'
jun 12 07:36:40 tee-fhaas systemd[1]: Started qgsd.service - Intel(R) TD Quoting Generation Service.
jun 12 07:36:40 tee-fhaas qgsd[16579]: Added signal handler
jun 12 07:36:40 tee-fhaas qgsd[16579]: About to create QgsServer with num_thread = 4
jun 12 07:36:40 tee-fhaas qgsd[16579]: About to start main loop
jun 12 07:37:26 tee-fhaas qgsd[16579]: About to close acceptor
jun 12 07:37:26 tee-fhaas systemd[1]: Stopping qgsd.service - Intel(R) TD Quoting Generation Servic>
jun 12 07:37:26 tee-fhaas qgsd[16579]: Stopped [0] connections, about to clear connection list
jun 12 07:37:26 tee-fhaas qgsd[16579]: Joined thread pool
jun 12 07:37:26 tee-fhaas qgsd[16579]: Stopped io_service
jun 12 07:37:26 tee-fhaas qgsd[16579]: Quit main loop
jun 12 07:37:26 tee-fhaas qgsd[16579]: Deleted QgsServer object
jun 12 07:37:26 tee-fhaas systemd[1]: qgsd.service: Deactivated successfully.
jun 12 07:37:26 tee-fhaas systemd[1]: Stopped qgsd.service - Intel(R) TD Quoting Generation Service.
-- Boot 364c6c4572104b99b2e4a667997c249a --
jun 12 07:38:51 tee-fhaas systemd[1]: Starting qgsd.service - Intel(R) TD Quoting Generation Servic>
jun 12 07:38:52 tee-fhaas systemd[1]: Started qgsd.service - Intel(R) TD Quoting Generation Service.
jun 12 07:38:52 tee-fhaas qgsd[2390]: Added signal handler
jun 12 07:38:52 tee-fhaas qgsd[2390]: About to create QgsServer with num_thread = 4
jun 12 07:38:52 tee-fhaas qgsd[2390]: About to start main loop
jun 12 08:11:28 tee-fhaas qgsd[2390]: Added a new connection, now [1]
jun 12 08:11:28 tee-fhaas qgsd[2390]: handle read header, status [system:0]
jun 12 08:11:28 tee-fhaas qgsd[2390]: Got header!
jun 12 08:11:28 tee-fhaas qgsd[2390]: body should be [1048] bytes!
jun 12 08:11:28 tee-fhaas qgsd[2390]: handle read body status [system:0]
jun 12 08:11:28 tee-fhaas qgsd[2390]: Got body!
jun 12 08:11:28 tee-fhaas qgsd[2390]: unpack message successfully in thread [73271033c740]
jun 12 08:11:28 tee-fhaas qgsd[2390]: call tee_att_create_context
jun 12 08:11:28 tee-fhaas qgsd[2390]: create context in thread[73270fa006c0]
jun 12 08:11:28 tee-fhaas qgsd[2390]: [QCNL] Error creating directory '/var/opt/qgsd/.dcap-qcnl/'.
jun 12 08:11:29 tee-fhaas qgsd[2390]: [QPL] No certificate data for this platform.
jun 12 08:11:29 tee-fhaas qgsd[2390]: [get_platform_quote_cert_data ../td_ql_logic.cpp:319] Error  returned from the p_sgx_get_quote_config API. 0xe011
jun 12 08:11:29 tee-fhaas qgsd[2390]: tee_att_init_quote return 0x11001
jun 12 08:11:29 tee-fhaas qgsd[2390]: resp_size is 0
jun 12 08:11:29 tee-fhaas qgsd[2390]: About to shutdown and close socket
jun 12 08:11:29 tee-fhaas qgsd[2390]: erased a connection, now [0]
```

Any idea what is happening?

**Comment by arvind5** [comment]
@diegoara96, the error message in QGS indicates the platform library doesn't have any platform cert data. Can you follow the steps posted on https://github.com/canonical/tdx/issues/131, especially the mpa_registration.log file? Did you see any failures?

**Comment by diegoara96** [comment]
> @diegoara96, the error message in QGS indicates the platform library doesn't have any platform cert data. Can you follow the steps posted on #131, especially the mpa_registration.log file? Did you see any failures?

 Thank you very much @arvind5 . The error was fixed as you indicated in issue #131 . All this happened when I updated the bios to the latest version and that also updated the TDX to the build recommended in this repo.

If anyone else encounters this problem, please check the SGX Auto MP Registration option, since in previous versions of the bios from the beginning of this year this option did not appear.

**Comment by hector-cao** [comment]
Thanks @arvind5 and @diegoara96 , closing
